include/rdma/fi_errno.h

```diff
@@ -194,6 +194,8 @@ enum {
 	FI_ENOAV	 = 267, /* Missing or unavailable address vector */
 	FI_EOVERRUN	 = 268, /* Queue has been overrun */
 	FI_ENORX	 = 269, /* Receiver not ready, no receive buffers available */
+	FI_EIOURING_PREP = 270, /* io_uring operation prepared, waiting for submission */
+	FI_EIOURING_FULL = 271, /* io_uring full */
```

Comment:
I'm reviewing patch by patch.  Do these really need to be exposed to the libfabric user?

Reply:
Is EIOURING_FULL equivalent to EBUSY or EAGAIN?

Reply:
`FI_EIOURING_PREP` is _not_ supposed to be forwarded to the libfabric user: it can remain private to the net provider. I see that `efa` defines its own set of errno codes in `prov/efa/src/efa_errno.h`. Do you want me to do the same for io_uring in the net provider?

As for `FI_EIOURING_FULL`, we can return `-FI_EOVERFLOW` and let the provider decide what's the best strategy: retry or fail. The `net` provider will have its own control flow mechanism to ensure that it never overflows the rings, so this error code is supposed to never happen. I think we can treat it as a critical error and disable the EP if it ever happens (likely because of a bug in libfabric or io_uring/liburing).